app/src/main/java/com/infomaniak/mail/ui/main/menu/MenuDrawerFragment.kt

```diff
@@ -17,15 +17,253 @@
+        manageAccount.setOnClickListener {
+            closeDrawer()
+            safeNavigate(
+                directions = ThreadListFragmentDirections.actionThreadListFragmentToManageMailAddressFragment(),
+                currentClassName = MenuDrawerFragment::class.java.name,
+            )
+        }
+        addAccount.setOnClickListener { startActivity(Intent(context, LoginActivity::class.java)) }
+        inboxFolder.setOnClickListener { openFolder(R.string.inboxFolder) }
+        customFolders.setOnClickListener {
+            customFoldersList.apply {
+                isVisible = !isVisible
+                expandCustomFolderButton.toggleChevron(!isVisible)
+            }
+        }
+        feedbacks.setOnClickListener {
+            closeDrawer()
+            context?.openUrl(BuildConfig.FEEDBACK_USER_REPORT)
+        }
+        help.setOnClickListener {
+            closeDrawer()
+            safeNavigate(
+                directions = ThreadListFragmentDirections.actionThreadListFragmentToHelpFragment(),
+                currentClassName = MenuDrawerFragment::class.java.name,
+            )
+        }
+        importMails.setOnClickListener {
+            closeDrawer()
+            // TODO: import mails
+        }
+        restoreMails.setOnClickListener {
+            closeDrawer()
+            // TODO: restore mails
+        }
+    }
+
+    private fun handleOnBackPressed() {
+        activity?.onBackPressedDispatcher?.addCallback(viewLifecycleOwner) {
+            if (isDrawerOpen?.invoke() == true) {
+                closeDrawer()
+            } else {
+                isEnabled = false
+                activity?.onBackPressed()
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        listenToCurrentMailbox()
+        listenToMailboxes()
+        listenToFolders()
+        listenToCurrentMailboxSize()
+    }
+
+    override fun onPause() {
+        currentMailboxJob?.cancel()
+        currentMailboxJob = null
+
+        mailboxesJob?.cancel()
+        mailboxesJob = null
+
+        foldersJob?.cancel()
+        foldersJob = null
+
+        super.onPause()
+    }
+
+    private fun listenToCurrentMailbox() {
+        if (currentMailboxJob != null) currentMailboxJob?.cancel()
+
+        currentMailboxJob = menuDrawerViewModel.viewModelScope.launch(Dispatchers.Main) {
+            MailData.currentMailboxFlow.filterNotNull().collect { currentMailbox ->
+                binding.accountSwitcherText.text = currentMailbox.email
+            }
+        }
+    }
+
+    private fun listenToMailboxes() = with(menuDrawerViewModel) {
+        if (mailboxesJob != null) mailboxesJob?.cancel()
+
+        mailboxesJob = viewModelScope.launch(Dispatchers.Main) {
+            uiMailboxesFlow.filterNotNull().collect { mailboxes ->
+                addressAdapter.setMailboxes(mailboxes.filterNot { it.mailboxId == AccountUtils.currentMailboxId })
+                addressAdapter.notifyDataSetChanged()
```

Comment:
To avoid duplication, it is the adapter that must notify itself of new additions and not the fragment